### ISSUE TITLE
syntax error, false positive

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1705]
```abap
DATA tab TYPE STANDARD TABLE OF i WITH DEFAULT KEY.
DATA row LIKE LINE OF tab.
DATA ref TYPE REF TO i.
APPEND INITIAL LINE TO tab REFERENCE INTO ref.
ref->* = 2.
LOOP AT tab INTO row.
  WRITE row.
ENDLOOP.
```

![image](https://user-images.githubusercontent.com/5888506/105683212-b0ba5f00-5ef3-11eb-9452-69f0e99a1149.png)
